spec.html

```diff
@@ -12491,6 +12494,29 @@ <h1>
       1. Return ~unused~.
     </emu-alg>
   </emu-clause>
+
+  <emu-clause id="sec-canbeheldweakly-abstract-operation" type="abstract operation">
+    <h1>
+      CanBeHeldWeakly (
+        _v_: unknown,
+      ): a Boolean
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>It returns *true* if and only if _v_ is considered to have an identity suitable for a weak reference. i.e. Is _v_ a valid candidate key of a WeakMap, or entry in a WeakSet, or target of a WeakRef or FinalizationRegistry.</dd>
+    </dl>
+    <emu-alg>
+      1. If Type(_v_) is Object, return *true*.
+      1. If Type(_v_) is Symbol, then
+        1. For each element _e_ of the GlobalSymbolRegistry List (see <emu-xref href="#sec-symbol.for"></emu-xref>), do
+          1. If SameValue(_e_.[[Symbol]], _v_) is *true*, return *false*.
+        1. Return *true*.
+      1. Return *false*.
+    </emu-alg>
+    <emu-note type="editor">
+      <p>Symbols in the GlobalSymbolRegistry (<emu-xref href="#sec-symbol.for"></emu-xref>) are not considered to have a suitable identity because they are permanently live. Well-known symbols (<emu-xref href="#sec-well-known-symbols"></emu-xref>) are also permanently live, but are nonetheless treated as suitable for weak reference because they are limited to elements of a fixed size list and therefore manageable by a variety of implementation approaches. However, using a well-known symbol as a key in a WeakMap is likely to &ldquo;leak&rdquo; memory resources in many implementations.</p>
```

Comment:
According to the 3rd bullet point under the definition of liveness, it's not that well-known symbols are "likely" to leak, it's that they will leak as specified.

Reply:
I think the term "leak" is vague enough that "likely" is justified. The value in the weakMap would be held onto until either the key is removed or until the weakMap itself is collected - and there are a small number of well-known symbols. So the leak is 'bound' in that regard. Compared to a leak in an app where a dialog is open and closed repeatedly and the memory can keep increasing until OOM; that style of leak feels different to the type of leak that could be described by using a well-known symbol as a weakMap key.

Reply:
Could we rephrase to something like: 
> However any values associated to a well-known symbol through a WeakMap will be kept live.

Reply:
Since well-known symbols are no longer explicitly live maybe the following wording is more appropriate.

```suggestion
      <p>Symbols in the GlobalSymbolRegistry (<emu-xref href="#sec-symbol.for"></emu-xref>) are not considered to have a suitable identity because they are permanently live. Well-known symbols (<emu-xref href="#sec-well-known-symbols"></emu-xref>) are likely to never be collected, but are nonetheless treated as suitable for weak reference because they are limited to elements of a fixed size list and therefore manageable by a variety of implementation approaches. However, any values associated to a well-known symbol in a live WeakMap is unlikely to be collected and will &ldquo;leak&rdquo; memory resources in many implementations.</p>
```

Reply:
Given the current definition of liveness, well-known symbols technically are not permanently live. It is possible to be in a state where the well-known symbols are not in the KeptAlive list and there does not exist a  future hypothetical WeakRef-oblivious execution that could observe the value of the symbol.

```js
// first run before all other code:
delete Symbol.prototype.constructor;
delete gbloalThis.Symbol;
delete Array.prototype.concat;
delete globalThis.ShadowRealm; // and all other ways to construct a new realm
delete globalThis.Proxy; // not needed but for good measure
// No longer an execution that could observe the identity of `Symbol.isConcatSpreadable`
```

That said I do not ever seeing engines putting in the work to try and collect them, it would be too small a benefit. I'll update the notes to make it clear that they are most likely always live, while engines are technically free to collect them if they so chose.

Reply:
I would say something like "are not considered suitable for use in WeakMaps or WeakRefs because such a symbol can be manifested without already having a reference to it", or something to that effect, rather than referring to liveness, collection, or identity.